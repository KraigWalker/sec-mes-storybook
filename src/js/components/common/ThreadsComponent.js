
import React from 'react';
import SecureMessageSummary from './SecureMessageSummary';
import TextArea from '../common/TextAreaComponent';

class ThreadsComponent extends React.Component {
   
    render() {
        const { ThreadDetail } = this.props;
        return (
           <li className="c-thread">
               <SecureMessageSummary message= { ThreadDetail } threadFlag={true} hasOnClick={false}/>
               <section>
                  {ThreadDetail.messageBody} 
               </section>
           </li>
        );
    }
};
export default ThreadsComponent;