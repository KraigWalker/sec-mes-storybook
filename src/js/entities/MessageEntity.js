import AccountEntity from './AccountEntity';
import {getISODateString} from '../utils/DateUtils';

/**
 * @class MessageEntity  Class for getter setter changes
 */
class MessageEntity {
    constructor() {
        this.id = null;
        this.account = new AccountEntity();
        this.reference = null;
        this.dateCreated = null;
        this.status = null;
        this.threadID = null;
        this.subject = null;
        this.message = null;
    }
    /**
     * @returns account mapped to message
     */
    getId() {
        return this.id;
    }
    /**
     * @returns account mapped to message
     */
    getAccount() {
        return this.account;
    }
    /**
     * @returns message reference
     */
    getReference() {
        return this.reference;
    }
    /**
     * @return Date Of message
     */
    getDateCreated() {
        return this.dateCreated;
    }
    /**
     * @return status
     */
    getStatus() {
        return this.status;
    }
    /**
     * @return threadID
     */
    getThreadID() {
        return this.threadID;
    }
    /**
     * @return subject of the message
     */
    getSubject() {
        return this.subject;
    }
    /**
     * @return message body
     */ 
    getMessageBody() {
        return this.message;
    }
    /**
     * 
     * @param {String} reference of the secure message
     */
    setId(id) {
        this.id = id; 
    }
    /**
     * 
     * @param {Object} Account Entity related to the message
     */
    setAccount(account) {
        if(account === null || account === undefined) {
        this.account.setId();
        this.account.setAccountNumber();
        } else {
            this.account.setId(account.id);
            this.account.setAccountNumber(account.number);
        }
    }
    /**
     * 
     * @param {String} reference of the secure message
     */
    setReference(reference) {
        this.reference = reference; 
    }
    /**
     * 
     * @param {*} date 
     */
    setDateCreated(date) {
        this.dateCreated = getISODateString(date);
    }
    /**
     * 
     * @param {String} status of the message
     */
    setStatus(status) {
        this.status = status;
    }
    /**
     * 
     * @param {*} Thread Id to link messages
     */
    setThreadId(id) {
        this.threadID = id;
    }
    /**
     * 
     * @param {*} subject of the message
     */
    setSubject(subject) {
        this.subject = subject;
    }
    /**
     * 
     * @param {*} message body
     */
    setMessageBody(message) {
        this.message = message
    }

}

export default MessageEntity;