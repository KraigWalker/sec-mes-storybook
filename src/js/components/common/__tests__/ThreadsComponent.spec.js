import React from 'react';
import { shallow, mount } from 'enzyme';
import { createMockStore } from 'redux-test-utils';
import ThreadsComponent from '../ThreadsComponent'
import { Provider } from "react-redux";
import MessageEntity from '../../../entities/MessageEntity';
jest.mock("../ThreadsComponent");

const mountWithStore = (component, store) => {
  const context = {store,};
  return mount(<Provider store={store}>
                    {component}
                </Provider>, { context });
};

describe('ThreadsComponent Component Check', () => {
  jest.fn(ThreadsComponent.prototype, 'getThreadsComponent');
  const messageEntity = new MessageEntity();
  const testState = {messages: {}};
  const store = createMockStore(testState);
  const componentWrap = mountWithStore(<ThreadsComponent ThreadDetail={messageEntity}/>, store);

  it("ThreadsComponent component should mount", () =>{
    expect(typeof componentWrap).toBe('object');
  });
  it("Props Check", () =>{
    expect(componentWrap.children().props().ThreadDetail).toEqual(messageEntity);
  });
  it("ThreadsComponent Check", () =>{
    expect(componentWrap.find('SecureMessageSummary')).toHaveLength(0);
  });
   it('should render correctly', () => {
   const tree = shallow(
     <ThreadsComponent />
   );
   expect(tree).toMatchSnapshot();
 });
});
