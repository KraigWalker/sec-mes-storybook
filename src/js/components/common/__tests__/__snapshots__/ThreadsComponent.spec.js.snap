// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadsComponent snapshot should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThreadsComponent
    ThreadDetail={
      Object {
        "message": Array [],
      }
    }
    content={Object {}}
    dispatch={[MockFunction]}
    messages={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(SecureMessageSummary)
          content={Object {}}
          hasOnClick={false}
          message={
            Object {
              "message": Array [],
            }
          }
          threadFlag={true}
        />,
        <pre
          className="c-thread__body"
        />,
      ],
      "className": "c-thread",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "content": Object {},
          "hasOnClick": false,
          "message": Object {
            "message": Array [],
          },
          "threadFlag": true,
        },
        "ref": null,
        "rendered": null,
        "type": [MockFunction],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "c-thread__body",
        },
        "ref": null,
        "rendered": Array [],
        "type": "pre",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(SecureMessageSummary)
            content={Object {}}
            hasOnClick={false}
            message={
              Object {
                "message": Array [],
              }
            }
            threadFlag={true}
          />,
          <pre
            className="c-thread__body"
          />,
        ],
        "className": "c-thread",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "content": Object {},
            "hasOnClick": false,
            "message": Object {
              "message": Array [],
            },
            "threadFlag": true,
          },
          "ref": null,
          "rendered": null,
          "type": [MockFunction],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "c-thread__body",
          },
          "ref": null,
          "rendered": Array [],
          "type": "pre",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
